internal/Panels/utils.js

```diff
@@ -0,0 +1,224 @@
+import {useRef, useState, useCallback, Children} from 'react';
+import hoc from '@enact/core/hoc';
+import EnactPropTypes from '@enact/core/internal/prop-types';
+import useChainRefs from '@enact/core/useChainRefs';
+import PropTypes from 'prop-types';
+import {createContext} from 'react';
+
+import useAutoFocus from './useAutoFocus';
+import useFocusOnTransition from './useFocusOnTransition';
+import useToggleRole from './useToggleRole';
+
+const WizardPanelsContext = createContext(null);
+
+// single-index ViewManagers need some help knowing when the transition direction needs to change
+// because the index is always 0 from its perspective.
+function useReverseTransition (index = -1, rtl) {
+	const prevIndex = useRef(index);
+	const reverse = useRef(rtl);
+	// If the index was changed, the panel transition is occured on the next cycle by `Panel`
+	const prev = {reverseTransition: reverse.current, prevIndex: prevIndex.current};
+
+	if (prevIndex.current !== index) {
+		reverse.current = rtl ? (index > prevIndex.current) : (index < prevIndex.current);
+		prevIndex.current = index;
+	}
+
+	return prev;
+}
+
+const defaultConfig = {
+	componentType: 'WizardPanels'
+};
+
+/**
+ * WizardPanelsRouter passes the children, footer, subtitle, and title from
+ * {@link sandstone/WizardPanels.Panel|WizardPanel} to
+ * {@link sandstone/WizardPanels.WizardPanelsBase|WizardPanelsBase}.
+ *
+ * @class WizardPanelsRouter
+ * @memberof sandstone/WizardPanels
+ * @hoc
+ * @private
+ */
+const WizardPanelsRouter = hoc(defaultConfig, (config, Wrapped) => {
```

Comment:
As we get component type by config, I think it would be better to rename it. Also, Please update utils.js file name.
```suggestion
const PanelsRouter = hoc(defaultConfig, (config, Wrapped) => {
```

Reply:
I changed the name.